server/tools/client_credentials_admin.py

```diff
@@ -0,0 +1,232 @@
+                + " all queues): "
+            )
+            max_priority[queue_name] = int(
+                input("Enter the max priority for this queue: ")
+            )
+        elif max_priority_input == "n":
+            break
+        else:
+            print("Invalid input. Please enter 'y' or 'n'.")
+    return max_priority
+
+
+def handle_allowed_queues_input() -> list:
+    """Gets a user inputted allowed_queues list"""
+    allowed_queues = []
+    while True:
+        allowed_queues_input = input(
+            "Do you want to specify a restricted queue that"
+            + " you are allowed to use? (y/n): "
+        )
+        if allowed_queues_input == "y":
+            queue_name = input("Enter the name of the queue: ")
+            allowed_queues.append(queue_name)
+        elif allowed_queues_input == "n":
+            break
+        else:
+            print("Invalid input. Please enter 'y' or 'n'.")
+    return allowed_queues
+
+
+def confirm_dialogue(entry: dict) -> bool:
+    """Prompts the user to confirm their changes"""
+    print(entry)
+    while True:
+        confirm = input("Confirm(y/n) that this entry looks correct: ")
+        if confirm == "y":
+            return True
+
+        if confirm == "n":
+            print("operation aborted")
+            return False
+
+        print("Enter y/n")
+
+
+def create_client_credential(db):
+    """
+    Creates new client_id, client_secret pair with max_priority
+    and allowed_queues
+    """
+    client_id = input("Enter the new client_id: ")
+    if (
+        db.client_permissions.count_documents(
+            {"client_id": client_id}, limit=1
+        )
+        == 1
+    ):
+        print("Client id already exists!")
+        return
+
+    client_secret = getpass.getpass("Enter the new client_secret: ")
+    client_secret_hash = bcrypt.hashpw(
+        client_secret.encode("utf-8"), bcrypt.gensalt()
+    ).decode()
+
+    max_priority = handle_max_priority_input()
+    allowed_queues = handle_allowed_queues_input()
+
+    entry = {
+        "client_id": client_id,
+        "max_priority": max_priority,
+        "allowed_queues": allowed_queues,
+    }
+    if confirm_dialogue(entry):
+        entry["client_secret_hash"] = client_secret_hash
+        db.client_permissions.insert_one(entry)
+        print("Entry has been created and stored in database")
+
+
+def edit_client_credential(db):
+    """
+    Edits both max_priority and allowed_queues given a client_id. This will
+    overwrite both of these lists in the database.
+    """
+
+    client_id = input("Enter the client_id you wish to edit: ")
+    max_priority = handle_max_priority_input()
+    allowed_queues = handle_allowed_queues_input()
+    confirm_output = confirm_dialogue(
+        {
+            "client_id": client_id,
+            "max_priority": max_priority,
+            "allowed_queues": allowed_queues,
+        }
+    )
+    if confirm_output:
+        db.client_permissions.find_one_and_update(
```

Comment:
There's no check to see if this actually updated something, and it didn't check that the client_id existed, so if you edit one that doesn't exist, then nothing happens but it also doesn't give you an error saying that it didn't exist